### PR TITLE
Change documentation

### PR DESCRIPTION
**Pull request by lassemand**
## Purpose

Faster local development setup

## Changes

Ensuring everything is setup using docker

### CODE REVIEW

File: test_exchange/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,19 @@
+FROM rust as rust-builder
+WORKDIR /app
+
+# Use cached dependencies builds as long as cargo files do not change
```
Comment (lassemand) [inline test_exchange/Dockerfile:4]:
```
❯ docker build .
[+] Building 4.2s (19/19) FINISHED
 => [internal] load .dockerignore                                                                                                                                                                                                                                          0.0s
 => => transferring context: 2B                                                                                                                                                                                                                                            0.0s
 => [internal] load build definition from Dockerfile                                                                                                                                                                                                                       0.0s
 => => transferring dockerfile: 640B                                                                                                                                                                                                                                       0.0s
 => [internal] load metadata for docker.io/library/debian:buster-slim                                                                                                                                                                                                      0.5s
 => [internal] load metadata for docker.io/library/rust:latest                                                                                                                                                                                                             0.7s
 => [internal] load build context                                                                                                                                                                                                                                          0.0s
 => => transferring context: 3.07kB                                                                                                                                                                                                                                        0.0s
 => [rust-builder  1/10] FROM docker.io/library/rust@sha256:6de6071df133f8be44dd4538c74e93590941c6d2b9c98853e05011714fbcf57d                                                                                                                                               0.0s
 => CACHED [stage-1 1/3] FROM docker.io/library/debian:buster-slim@sha256:8e5da7fd9aef8c0b3f23bb8c3f391667b50929984b7daa666ea1c7730f3b73b9                                                                                                                                 0.0s
 => CACHED [rust-builder  2/10] WORKDIR /app                                                                                                                                                                                                                               0.0s
 => CACHED [rust-builder  3/10] COPY ./Cargo.toml .                                                                                                                                                                                                                        0.0s
 => CACHED [rust-builder  4/10] COPY ./Cargo.lock .                                                                                                                                                                                                                        0.0s
 => CACHED [rust-builder  5/10] RUN mkdir -p ./src && echo 'fn main() { println!("Dummy!"); }' > ./src/main.rs                                                                                                                                                             0.0s
 => CACHED [rust-builder  6/10] RUN cargo build --release                                                                                                                                                                                                                  0.0s
 => CACHED [rust-builder  7/10] RUN rm -rf ./src                                                                                                                                                                                                                           0.0s
 => [rust-builder  8/10] COPY ./src ./src                                                                                                                                                                                                                                  0.0s
 => [rust-builder  9/10] RUN touch -a -m ./src/main.rs                                                                                                                                                                                                                     0.2s
 => [rust-builder 10/10] RUN cargo build --release                                                                                                                                                                                                                         3.3s
 => [stage-1 2/3] COPY --from=rust-builder /app/target/release/eur2ccd-service-test-exchange /usr/local/bin/                                                                                                                                                               0.0s
 => [stage-1 3/3] WORKDIR /usr/local/bin                                                                                                                                                                                                                                   0.0s
 => exporting to image                                                                                                                                                                                                                                                     0.0s
 => => exporting layers                                                                                                                                                                                                                                                    0.0s
 => => writing image sha256:934b2352a3b12246378fdd014c13903834ce3e93db64aed2e4c8a2324f725cab
```

Comment (lassemand) [reply test_exchange/Dockerfile:4]: This is while refactoring code only

File: scripts/debian-package/deb.Dockerfile [hunk 2]
```diff
@@ -11,8 +11,6 @@ WORKDIR /build
 
 COPY . /build
 
-RUN rustup component add rustfmt
```
Comment (lassemand) [inline scripts/debian-package/deb.Dockerfile:14]: Why do we install this when it is not used?

File: test_exchange/Dockerfile (hunk 1 above)
Comment (lassemand) [reply test_exchange/Dockerfile:4]:
```
❯ docker build .
[+] Building 57.3s (19/19) FINISHED
 => [internal] load .dockerignore                                                                                                                                                                                                                                          0.0s
 => => transferring context: 2B                                                                                                                                                                                                                                            0.0s
 => [internal] load build definition from Dockerfile                                                                                                                                                                                                                       0.0s
 => => transferring dockerfile: 599B                                                                                                                                                                                                                                       0.0s
 => [internal] load metadata for docker.io/library/debian:buster-slim                                                                                                                                                                                                      2.2s
 => [internal] load metadata for docker.io/library/rust:latest                                                                                                                                                                                                             2.1s
 => [rust-builder  1/10] FROM docker.io/library/rust@sha256:6de6071df133f8be44dd4538c74e93590941c6d2b9c98853e05011714fbcf57d                                                                                                                                              13.4s
 => => resolve docker.io/library/rust@sha256:6de6071df133f8be44dd4538c74e93590941c6d2b9c98853e05011714fbcf57d                                                                                                                                                              0.0s
 => => sha256:4ef2b16cab43e396a327507ff738e3a35e5fc7819aad3f4aef7ac992289cc1b6 6.11kB / 6.11kB                                                                                                                                                                             0.0s
 => => sha256:df2021ddb7d686bdbb125598b2a6163d63035f080356b3014595f354ea0b40d6 49.61MB / 49.61MB                                                                                                                                                                           1.7s
 => => sha256:8d647f1dd7e741209a8a75083ccc889e39cb3e94c17f45441eae96e1a679d971 23.58MB / 23.58MB                                                                                                                                                                           1.2s
 => => sha256:6de6071df133f8be44dd4538c74e93590941c6d2b9c98853e05011714fbcf57d 988B / 988B                                                                                                                                                                                 0.0s
 => => sha256:18bba517f7738b8c721c7339bd834a35970333537e6b50a3407a781765916612 1.38kB / 1.38kB                                                                                                                                                                             0.0s
 => => sha256:5cdd9a70365f741a6b9f7a4e32cdb7d4aa29ac73da0b78ca0a83e937f285fdd5 63.99MB / 63.99MB                                                                                                                                                                           4.3s
 => => sha256:95089c600b361807380090316c250b0b8eaf4fa2175b11ac8f49bb7581c61125 202.45MB / 202.45MB                                                                                                                                                                         7.0s
 => => extracting sha256:df2021ddb7d686bdbb125598b2a6163d63035f080356b3014595f354ea0b40d6                                                                                                                                                                                  1.0s
 => => sha256:409bf65580386de8886329931c5566015d671b2c917cb4a9266d70342c3c65f6 249.95MB / 249.95MB                                                                                                                                                                        10.1s
 => => extracting sha256:8d647f1dd7e741209a8a75083ccc889e39cb3e94c17f45441eae96e1a679d971                                                                                                                                                                                  0.3s
 => => extracting sha256:5cdd9a70365f741a6b9f7a4e32cdb7d4aa29ac73da0b78ca0a83e937f285fdd5                                                                                                                                                                                  1.0s
 => => extracting sha256:95089c600b361807380090316c250b0b8eaf4fa2175b11ac8f49bb7581c61125                                                                                                                                                                                  2.8s
 => => extracting sha256:409bf65580386de8886329931c5566015d671b2c917cb4a9266d70342c3c65f6                                                                                                                                                                                  2.9s
 => [stage-1 1/3] FROM docker.io/library/debian:buster-slim@sha256:8e5da7fd9aef8c0b3f23bb8c3f391667b50929984b7daa666ea1c7730f3b73b9                                                                                                                                        2.2s
 => => resolve docker.io/library/debian:buster-slim@sha256:8e5da7fd9aef8c0b3f23bb8c3f391667b50929984b7daa666ea1c7730f3b73b9                                                                                                                                                0.0s
 => => sha256:8e5da7fd9aef8c0b3f23bb8c3f391667b50929984b7daa666ea1c7730f3b73b9 984B / 984B                                                                                                                                                                                 0.0s
 => => sha256:8ab68822154393a571712e420acdf94ea11bee6de4192341699c4059a91d1225 529B / 529B                                                                                                                                                                                 0.0s
 => => sha256:e995a520cb24e6d55035acaa5330585c8cbe14093ba35867e97b2fa269b21e97 1.48kB / 1.48kB                                                                                                                                                                             0.0s
 => => sha256:78b429e3efdd9f2b163abf0bca6b1462a25b1c5eddc1dd47f5d3445c6413cfa1 25.97MB / 25.97MB                                                                                                                                                                           1.4s
 => => extracting sha256:78b429e3efdd9f2b163abf0bca6b1462a25b1c5eddc1dd47f5d3445c6413cfa1                                                                                                                                                                                  0.6s
 => [internal] load build context                                                                                                                                                                                                                                          0.0s
 => => transferring context: 44.63kB                                                                                                                                                                                                                                       0.0s
 => [rust-builder  2/10] WORKDIR /app                                                                                                                                                                                                                                      0.1s
 => [rust-builder  3/10] COPY ./Cargo.toml .                                                                                                                                                                                                                               0.0s
 => [rust-builder  4/10] COPY ./Cargo.lock .                                                                                                                                                                                                                               0.0s
 => [rust-builder  5/10] RUN mkdir -p ./src && echo 'fn main() { println!("Dummy!"); }' > ./src/main.rs                                                                                                                                                                    0.1s
 => [rust-builder  6/10] RUN cargo build --release                                                                                                                                                                                                                        37.3s
 => [rust-builder  7/10] RUN rm -rf ./src                                                                                                                                                                                                                                  0.3s
 => [rust-builder  8/10] COPY ./src ./src                                                                                                                                                                                                                                  0.0s
 => [rust-builder  9/10] RUN touch -a -m ./src/main.rs                                                                                                                                                                                                                     0.3s
 => [rust-builder 10/10] RUN cargo build --release                                                                                                                                                                                                                         3.4s
 => [stage-1 2/3] COPY --from=rust-builder /app/target/release/eur2ccd-service-test-exchange /usr/local/bin/                                                                                                                                                               0.0s
 => [stage-1 3/3] WORKDIR /usr/local/bin                                                                                                                                                                                                                                   0.0s
 => exporting to image                                                                                                                                                                                                                                                     0.0s
 => => exporting layers                                                                                                                                                                                                                                                    0.0s
 => => writing image sha256:ac07261ca1c4e754b2a26392f5996094d78f6da1bf679c4866418547afa502c0
```

Comment (lassemand) [reply test_exchange/Dockerfile:4]: This is while modifying cargo files

**Comment by abizjak** [comment]
Can you explain what problem you are solving here? Local development is very simple already by just running cargo build.

Docker just adds latency and complexity. What is the point of this?

**Comment by lassemand** [comment]
> Can you explain what problem you are solving here? Local development is very simple already by just running cargo build.
> 
> Docker just adds latency and complexity. What is the point of this?

I had to test whether the code was mysql 8 compatible. Hence I had to setup the external dependencies (reporter node, database, exchange) to ensure I could bump the database. 

cargo run does not setup external dependencies. It just loads up an instance of the application.

The purpose of this is being helpful to my fellow developers setting up a local instance of the application using only a single command when entering the repository.

**Pull request closed without merging**